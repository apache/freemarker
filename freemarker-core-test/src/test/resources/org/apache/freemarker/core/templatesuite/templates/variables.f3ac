[#ftl]
[#--
  Licensed to the Apache Software Foundation (ASF) under one
  or more contributor license agreements.  See the NOTICE file
  distributed with this work for additional information
  regarding copyright ownership.  The ASF licenses this file
  to you under the Apache License, Version 2.0 (the
  "License"); you may not use this file except in compliance
  with the License.  You may obtain a copy of the License at

    http://www.apache.org/licenses/LICENSE-2.0

  Unless required by applicable law or agreed to in writing,
  software distributed under the License is distributed on an
  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
  KIND, either express or implied.  See the License for the
  specific language governing permissions and limitations
  under the License.
--]
<html>
<head>
<title>FreeMarker: Variable Test</title>
</head>
<body>
[#assign list = ["one", "two", "three", "four", "five"]]
[#assign hash = {"output" : "My message.", "key" : list}]
[#assign hash2 = {"value" : hash}]
[#assign items = {"mykey" : "key", "_test", "out"}]

<p>A simple test follows:</p>

<p>${message}</p>

<p>Now get into variable nesting:</p>

<p>${hash.output}</p>
<p>${hash["output"]}</p>
<p>${hash. output}</p>
<p>${hash .output}</p>
<p>${hash 
    .output}</p>
<p>${hash 
    . output}</p>
<p>${hash ["output"]}</p>
<p>${hash
    [ "output" ]}</p>

<p>More deep nesting...</p>

<p>${hash2.value.output}</p>
<p>${hash2.value.key[0]}</p>
<p>${hash2["value"]["key"][0]}</p>


<p>Nesting inside nesting...</p>

<p>${hash2.value[ items.mykey ][ 1 ]}</p>
<p>${hash2.value[ items[ "mykey" ]][ 1 ]}</p>
<p>${hash2.value[ items[ "my" + items.mykey ]][ 1 ]}</p>
<p>${hash2.value[ items[ "my" + items["mykey"] ]][ 1 ]}</p>

<p>Test underscores...</p>

<p>${items[ "_test" ]}</p>
<p>${items._test}</p>

${"God save the queen."?wordList[1]?upperCase}

</body>
</html>
