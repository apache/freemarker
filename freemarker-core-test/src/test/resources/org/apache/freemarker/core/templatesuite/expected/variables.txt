/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *   http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing,
 * software distributed under the License is distributed on an
 * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 * KIND, either express or implied.  See the License for the
 * specific language governing permissions and limitations
 * under the License.
 */
<html>
<head>
<title>FreeMarker: Variable Test</title>
</head>
<body>

<p>A simple test follows:</p>

<p>Hello, world!</p>

<p>Now get into variable nesting:</p>

<p>My message.</p>
<p>My message.</p>
<p>My message.</p>
<p>My message.</p>
<p>My message.</p>
<p>My message.</p>
<p>My message.</p>
<p>My message.</p>

<p>More deep nesting...</p>

<p>My message.</p>
<p>one</p>
<p>one</p>


<p>Nesting inside nesting...</p>

<p>two</p>
<p>two</p>
<p>two</p>
<p>two</p>

<p>Test underscores...</p>

<p>out</p>
<p>out</p>

SAVE

</body>
</html>
